### ISSUE TITLE
`next` does not preserve the POST params for other routes

### ISSUE THREAD

**Issue by nikita-volkov** [scotty-web/scotty#147]
I had the following code to check, whether the user has access, until I found out that currently it's impossible to use forms with such an approach

``` haskell
    matchAny (regex "^/admin") $ do
      id <- fmap fst $ sessionState State.get
      case id of
        Nothing -> redirect "/login"
        Just id -> do
          adminAccess <- tx Nothing $ DB.adminAccess id
          unless adminAccess $ raise Unauthorised
          next
```

**Comment by xich** [comment]
Ugh, thanks for reporting. I've got a rewrite of the way request bodies are handled that is about 80% done and should solve this. Unfortunately I'm on another deadline, so not likely to get it finished until next week.

**Comment by d12frosted** [comment]
The same here. Just wanted to add some logging for debug purposes.

**Comment by thomasjm** [comment]
Hello, is this bug still outstanding? I have a mysterious issue that would be explained by this...

**Comment by thomasjm** [comment]
Aha, I just confirmed that using `next` makes the **body** of a POST request disappear for subsequent routes :(

This is a big problem for me, did your rewrite ever get landed @xich ?

**Comment by ocramz** [comment]
candidate fix by @thomasjm : https://github.com/scotty-web/scotty/pull/206/files

**Comment by ocramz** [comment]
part of the fix could be to move ActionEnv inside a state-like variable : https://github.com/scotty-web/scotty/issues/131#issuecomment-63485982

**Comment by ocramz** [comment]
Fixed by #308
